### DISCUSSION TITLE
SellingApiBadRequestException for RDT

### DISCUSSION THREAD

**Discussion by sktong** [saleweaver/python-amazon-sp-api#447] [Q&A]
Hi, 

I am trying to use the following block of code to get my restricted data token:

res = Tokens(credentials=credentials).create_restricted_data_token(restrictedResources=[
     {
         "method": "GET",
         "path": "/orders/v0/orders",
         "dataElements": ["buyerInfo", "shippingAddress"]
     }
  ])

But it comes up with the following errors:
SellingApiBadRequestException: [{'code': 'InvalidInput', 'message': 'Application does not have access to one or more requested data elements: [buyerInfo, shippingAddress]', 'details': ''}]

I just the copy and paste the code from https://sp-api-docs.saleweaver.com/endpoints/tokens/ but I don't know why I got this errors. Could you please help me solve this? Thanks so much.

**Comment by zonwizard** [comment]
even though restrictedResources parameter is documented as an Array, it seems to be a single object.
Try this:

res = Tokens(credentials=credentials).create_restricted_data_token(restrictedResources=
    {
    "method": "GET",
    "path": "/orders/v0/orders",
    "dataElements": ["buyerInfo", "shippingAddress"]
    }
)

**Reply by sktong** [reply]
Thanks. It seems that I need to adjustment my developer profile and apply for the access to the restricted data first.
